docs/admin/Offline_System_Certificate_Renewal.md

```diff
@@ -6,65 +6,50 @@ Offline System Certificate Renewal
+### IPA Environment (Uses LDAPI)
 
-    ````
-    # pki-server cert-find
+#### Assumptions:
 
-      Cert ID: ca_signing
-      Nickname: ca_signing
-      Serial Number: 0x1
-      Subject DN: CN=CA Signing Certificate,OU=pki-tomcat,O=EXAMPLE
-      Issuer DN: CN=CA Signing Certificate,OU=pki-tomcat,O=EXAMPLE
-      Not Valid Before: Wed Dec 19 17:33:21 2018
-      Not Valid After: Sun Dec 19 17:33:21 2038
-    ````
+- Valid CA certificate [WIP to remove this assumption]
+- `cert-fix` must be run as `root`
+- `LDAPI` must be configured, with `root` autobinding to `cn=Directory Manager` or other account with privileges on `o=ipaca` subtree, including password reset privileges
+- The password of the specified agent account will be reset. If needed, it can be changed back afterwards (manually; successful execution of `cert-fix` proves that the operator has privileges to do this)
+- If Dogtag was configured to use TLS certificate authentication to bind to LDAP, the password on the `pkidbuser` account will be reset. (If password authentication was already used, the password does not get reset)
+- LDAPI (ldappasswd) and need to be root
 
-2. Check details of the admin certificate
+#### Usage:
 
-    ````
-    # certutil -L \
-    -d <admin NSS database> \
-    -n <admin cert nickname>
+One line tool that fixes **all** certificates:
 
-    Certificate:
-        Data:
-            Version: 3 (0x2)
-            Serial Number: 6 (0x6)
-            Signature Algorithm: PKCS #1 SHA-256 With RSA Encryption
-            Issuer: "CN=CA Signing Certificate,OU=pki-tomcat,O=EXAMPLE"
-            Validity:
-                Not Before: Sat Dec 15 02:16:26 2018
-                Not After : Fri Dec 04 02:16:26 2020
-            Subject: "CN=PKI Administrator,E=caadmin@example.com,OU=pki-tomcat,O=
-                EXAMPLE"
+    # pki-server cert-fix \
+    --agent-uid <admin UID> \
+    --ldapi-socket <Directory Server LDAPI Socket's path>
 
-    ````
+If you need to fix only a **specific system certificates**, use the `--cert <Cert_ID>` option. If you need to renew **non-system certs**, use the `--extra-cert <Serial>` option.
 
-## Automated Renewal Process
 
-One line tool that fixes all certificates:
+### Standalone PKI environment (Uses LDAPS)
 
-    # pki-server cert-fix \
-    -n <admin cert nickname> \
-    -d <admin NSS database> \
-    -c <admin NSS database password>
+#### Assumptions:
 
-One line tool to fix one particular certificate:
+- Valid CA certificate [WIP to remove this assumption]
+- TLS configured Directory Server
+- If Dogtag was configured to use TLS certificate authentication to bind to LDAP, a Valid DS service certificate
 
-    # pki-server cert-fix --cert <cert ID> \
-    -n <admin cert nickname> \
-    -d <admin NSS database> \
-    -c <admin NSS database password>
+#### Usage:
+
+One line tool that fixes all certificates:
+
+    # pki-server cert-fix \
+    --ldap-url <LDAP URL> \
```

Comment:
Is the `--ldap-url` param optional? It probably can be obtained from CS.cfg.

Reply:
It's feasible but IMO if they're already doing cert-fix, this is not a big ask.  In any case this PR is to update documentation to match the current implementation so we can have a debate about default `--ldap-url` somewhere else ;)